tests/Mocks/ClusterInfoMock.cpp

```diff
@@ -0,0 +1,989 @@
+          result = result.mapError(appendErrorMessage);
+        }
+        return result;
+      });
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief create collection in coordinator, the return value is an ArangoDB
+/// error code and the errorMsg is set accordingly. One possible error
+/// is a timeout, a timeout of 0.0 means no timeout.
+////////////////////////////////////////////////////////////////////////////////
+Result ClusterInfo::createCollectionCoordinator(  // create collection
+    std::string const& databaseName, std::string const& collectionID,
+    uint64_t numberOfShards, uint64_t replicationFactor, uint64_t writeConcern,
+    bool waitForReplication,
+    velocypack::Slice json,  // collection definition
+    double timeout,          // request timeout,
+    bool isNewDatabase,
+    std::shared_ptr<LogicalCollection> const& colToDistributeShardsLike,
+    replication::Version replicationVersion) {
+  TRI_ASSERT(ServerState::instance()->isCoordinator());
+  auto serverState = ServerState::instance();
+  std::vector<ClusterCollectionCreationInfo> infos{
+      ClusterCollectionCreationInfo{
+          collectionID, numberOfShards, replicationFactor, writeConcern,
+          waitForReplication, json, serverState->getId(),
+          serverState->getRebootId()}};
+  double const realTimeout = getTimeout(timeout);
+  double const endTime = TRI_microtime() + realTimeout;
+  return createCollectionsCoordinator(databaseName, infos, endTime,
+                                      isNewDatabase, colToDistributeShardsLike,
+                                      replicationVersion);
+}
+
+Result ClusterInfo::createCollectionsCoordinator(
+    std::string const& databaseName,
+    std::vector<ClusterCollectionCreationInfo>& infos, double endTime,
+    bool isNewDatabase,
+    std::shared_ptr<const LogicalCollection> const& colToDistributeShardsLike,
+    replication::Version replicationVersion) {
+  TRI_ASSERT(ServerState::instance()->isCoordinator());
+  using arangodb::velocypack::Slice;
+
+  LOG_TOPIC("98761", DEBUG, Logger::CLUSTER)
+      << "Starting createCollectionsCoordinator for " << infos.size()
+      << " collections in database " << databaseName
+      << " isNewDatabase: " << isNewDatabase
+      << " first collection name: " << infos[0].name;
+
+  // The following three are used for synchronization between the callback
+  // closure and the main thread executing this function. Note that it can
+  // happen that the callback is called only after we return from this
+  // function!
+  auto dbServerResult =
+      std::make_shared<std::atomic<std::optional<ErrorCode>>>(std::nullopt);
+  auto nrDone = std::make_shared<std::atomic<size_t>>(0);
+  auto errMsg = std::make_shared<std::string>();
+  auto cacheMutex = std::make_shared<std::mutex>();
+  auto cacheMutexOwner = std::make_shared<std::atomic<std::thread::id>>();
+  auto isCleaned = std::make_shared<bool>(false);
```

Comment:
Note: these changes will conflict with an open PR from @MBkkt : https://github.com/arangodb/arangodb/pull/19503

Reply:
Thanks for this 👍 Adapted the code to the new open PR.